net/tstun/wrap.go

```diff
@@ -566,14 +713,16 @@ func (t *Wrapper) Read(buffs [][]byte, sizes []int, offset int) (int, error) {
 	defer parsedPacketPool.Put(p)
 	for _, data := range res.data {
 		p.Decode(data[res.dataOffset:])
+		if capt := t.captureHook.Load(); capt != nil {
+			capt(capture.FromLocal, time.Now(), data[res.dataOffset:])
+		}
+
+		t.snatV4(p)
```

Comment:
A bit further down we do:

```
n := copy(buffs[buffsPos][offset:], data[res.dataOffset:])
```

Now that `packet.Parsed` is also responsible for mutating, can we not access the underlying slice but instead do `p.Buffer()` ?

Reply:
Done